### ISSUE TITLE
Implementing Nested Menu Dropdowns

### ISSUE THREAD

**Issue by abumere** [meanjs/mean#1636]
Is there currently a way to implement a nested menu dropdown? 

As in: 

Menu Title (dropdown)
-- Sub Menu Item (Dropdown)
----Menu item (link)

**Comment by hyperreality** [comment]
Sorry to give an unhelpful answer... but I really wouldn't be a fan of using this, design-wise.

Anyway, you should probably ask this question on Bootstrap forums, it's a question about their navbar, not about the MEAN project.

**Comment by mleanos** [comment]
@abumere @hyperreality The [menuService](https://github.com/meanjs/mean/blob/master/modules/core/client/services/menu.client.service.js#L75) supports nested items. 

@abumere Take a look at that to see if it would suit your needs. If not, then can you describe your requirement a bit more?

**Comment by hyperreality** [comment]
@mleanos The way I interpreted it, he wants a second level of nesting.

**Comment by mleanos** [comment]
@hyperreality I think that's possible with the `addSubMenuItem` method of the *menuService*.

**Comment by hyperreality** [comment]
@mleanos Unless I'm being stupid, `addSubMenuItem` is one level of nesting less than he wants. For instance, the articles menu has the "List Articles" submenuitem, but I believe he would want a second dropdown coming out of that.

`addSubMenuItem` takes the `parentItemState` parameter, and if you set this to a submenu state rather than the navbar menu state, it doesn't work:

`menuService.addSubMenuItem('topbar', 'articles.list', { // Doesn't work`

I believe that's because it's not supported by Bootstrap.

**Comment by mleanos** [comment]
@hyperreality You're right. The data structure for the Menus can support this, but the functionality within the Menus service, and header template doesn't. Updating the `addSubMenuItem` method to find a nested Menu item based on the parent state, would work to add the menu item as a child to the desired menu item. However, then we would have to update the header template to accommodate the possibility of a nested structure.

As it stands though, this isn't supported but I don't see why the needs couldn't be satisfied with a reasonable amount of effort.

**Comment by mleanos** [comment]
@abumere Something to consider: https://github.com/twbs/bootstrap/pull/6342#issuecomment-11594010

You can probably figure out a way to achieve this, but it would most likely be something that you had to hack together, as it's not generally supported/recommended.
